iceoryx2/src/node/mod.rs

```diff
@@ -0,0 +1,709 @@
+                        "{} since an internal failure occurred while acquiring the node state.", msg);
+            }
+        }
+    }
+
+    fn open_node_storage(
+        config: &Config,
+        node_name: &FileName,
+    ) -> Result<Option<Service::StaticStorage>, NodeReadStorageFailure> {
+        let details_config = node_details_config::<Service>(config, node_name);
+        let result = <Service::StaticStorage as StaticStorage>::Builder::new(
+            &FileName::new(b"node").unwrap(),
+        )
+        .config(&details_config)
+        .has_ownership(false)
+        .open();
+
+        if let Ok(result) = result {
+            return Ok(Some(result));
+        }
+
+        let msg = "Unable to open node config storage";
+        let origin = format!("open_node_storage({:?}, {:?})", config, node_name);
+
+        match result.err().unwrap() {
+            StaticStorageOpenError::DoesNotExist => Ok(None),
+            StaticStorageOpenError::Read => {
+                fail!(from origin, with NodeReadStorageFailure::ReadError,
+                    "{} since the node config storage could not be read.", msg);
+            }
+            StaticStorageOpenError::IsLocked => {
+                fail!(from origin, with NodeReadStorageFailure::Corrupted,
+                    "{} since the node config storage seems to be uninitialized but the state should always be present.", msg);
+            }
+            StaticStorageOpenError::InternalError => {
+                fail!(from origin, with NodeReadStorageFailure::InternalError,
+                    "{} due to an internal failure while opening the node config storage.", msg);
+            }
+        }
+    }
+
+    fn get_node_details(
+        config: &Config,
+        node_name: &FileName,
+    ) -> Result<Option<NodeDetails>, NodeReadStorageFailure> {
+        let node_storage = Self::open_node_storage(config, node_name)?;
+        if node_storage.is_none() {
+            return Ok(None);
+        }
+        let node_storage = node_storage.unwrap();
+
+        let mut read_content =
+            String::from_utf8(vec![b' '; node_storage.len() as usize]).expect("");
+
+        let origin = format!("get_node_details({:?}, {:?})", config, node_name);
+        let msg = "Unable to read node details";
+
+        if node_storage
+            .read(unsafe { read_content.as_mut_vec() }.as_mut_slice())
+            .is_err()
+        {
+            fail!(from origin, with NodeReadStorageFailure::ReadError,
+                "{} since the content of the node config storage could not be read.", msg);
+        }
+
+        let node_details = fail!(from origin,
+                    when Service::ConfigSerializer::deserialize::<NodeDetails>(unsafe { read_content.as_mut_vec()}),
+                    with NodeReadStorageFailure::Corrupted,
+                "{} since the contents of the node config storage is corrupted.", msg);
+
+        Ok(Some(node_details))
```

Comment:
Food for thought to make the error handling a bit more idiomatic
```rs
Ok(Some(
    Service::ConfigSerializer::deserialize::<NodeDetails>(unsafe { read_content.as_mut_vec() })
        .map_err_with_ctx(origin, |_| {
            fail2!(
                NodeReadStorageFailure::Corrupted,
                "{} since the contents of the node config storage is corrupted.",
                msg
            )
        })?
))
```

Reply:
I have to admit that I find the "idiomatic" way a lot harder to read than the current version, therefore I would keep mine.

Reply:
For me it's the other way around. It's quite hard to read the code and I often have to read it multiple times to untangle the actual code flow.

Well, nothing to discuss here. I have to think about other options first.

Reply:
Then we have to find an approach that makes it easy to read for everyone.